src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java

```diff
@@ -82,7 +82,7 @@
 public class JRuleEngine implements PropertyChangeListener {
     private static final String[] EMPTY_LOG_TAGS = new String[0];
     private static final int AWAIT_TERMINATION_THREAD_SECONDS = 2;
-    private List<JRuleExecutionContext> contextList = new ArrayList<>();
+    private List<JRuleExecutionContext> contextList = new CopyOnWriteArrayList<>();
```

Comment:
nice, i didn't know this class, just Collections.synchronizedList which does not synchronize the iterator.